framework/mesh/LogicalVolume/LogicalVolumeInterface.cc

```diff
@@ -0,0 +1,38 @@
+#include "LogicalVolumeInterface.h"
+
+#include "LogicalVolume.h"
+
+#include "chi_runtime.h"
+
+namespace chi_mesh
+{
+
+chi::InputParameters LogicalVolumeInterface::GetInputParameters()
+{
+  chi::InputParameters params;
+
+  params.AddOptionalParameter(
+    "logical_volume", 0, "Handle to a logical_volume.");
+
+  return params;
+}
+
+LogicalVolumeInterface::LogicalVolumeInterface(
+  const chi::InputParameters& params)
+  : logical_volume_(params.ParametersAtAssignment().Has("logical_volume")
+                      ? Chi::GetStackItemPtrAsType<const LogicalVolume>(
+                          Chi::object_stack,
+                          params.GetParamValue<size_t>("logical_volume"),
+                          __FUNCTION__)
+                      : nullptr)
+{
+}
+
+const LogicalVolume* LogicalVolumeInterface::GetLogicalVolume() const
+{
+  if (not logical_volume_) return &(*logical_volume_);
+
+  return nullptr;
```

Comment:
```suggestion
 return not logical_volume_ ? nullptr : &(*logical_volume_);
```
This logic seems wrong

Reply:
Modified slightly but done